src/core/sock/sockinfo_ulp.cpp

```diff
@@ -116,10 +116,34 @@ template <typename T> static void dlsym_default(T &ptr, const char *name)
     dlsym_handle(ptr, name, RTLD_DEFAULT);
 }
 
-#define XLIO_TLS_API_FIND(__name) dlsym_default(s_tls_api.__name, #__name);
+static void *openssl_handle = nullptr;
 
-void xlio_tls_api_setup()
+#define XLIO_TLS_API_FIND(__name) dlsym_handle(s_tls_api.__name, #__name, openssl_handle);
+
+inline bool check_openssl_loaded()
+{
+    openssl_handle = dlopen("libssl.so.3", RTLD_NOW | RTLD_GLOBAL);
```

Comment:
The name of the library can change in the future. at least it should be some changeable define somewhere